cherab/core/laser/node.pyx

```diff
@@ -0,0 +1,233 @@
+from raysect.primitive cimport Cylinder
+from raysect.optical cimport World, AffineMatrix3D, Primitive, Ray
+from raysect.optical.material.emitter.inhomogeneous cimport NumericalIntegrator
+from raysect.core cimport translate, Material
+
+from cherab.core.laser.material cimport LaserMaterial
+from cherab.core.laser.scattering cimport LaserEmissionModel
+from cherab.core.laser.models.laserspectrum_base import LaserSpectrum
+from cherab.core.utility import Notifier
+from libc.math cimport M_PI
+
+from math import ceil
+
+cdef double DEGREES_TO_RADIANS = (M_PI / 180)
+
+
+cdef class ModelManager:
+
+    def __init__(self):
+        self._models = []
+        self.notifier = Notifier()
+
+    def __iter__(self):
+        return iter(self._models)
+
+    cpdef object set(self, object models):
+
+        # copy models and test it is an iterable
+        models = list(models)
+
+        # check contents of list are laser models
+        for model in models:
+            if not isinstance(model, LaserEmissionModel):
+                raise TypeError('The model list must consist of only LaserEmissionModel objects.')
+
+        self._models = models
+        self.notifier.notify()
+
+    cpdef object add(self, LaserEmissionModel model):
+
+        if not model:
+            raise ValueError('Model must not be None type.')
+
+        self._models.append(model)
+        self.notifier.notify()
+
+    cpdef object clear(self):
+        self._models = []
+        self.notifier.notify()
+
+cdef class Laser(Node):
+
+    def __init__(self, double length=1, double radius=0.05, object parent=None, AffineMatrix3D transform=None,
+                 double importance=0., str name=None):
+
+        super().__init__(parent, transform, name)
+
+        # set init values of the laser
+        self._set_init_values()
+
+        # change reporting and tracking
+        self.notifier = Notifier()
+        self.notifier.add(self._configure_geometry)
+
+        #setup model manager
+        self._models = ModelManager()
```

Comment:
I added self._configure_geometry to manager notifier

Reply:
I moved the geometry from the Laser node to the LaserProfile class, this will allow to do more effective geometries for the laser shapes in the future.